v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/endtoend/ForwardAndReverseMigrationEndToEndIT.java

```diff
@@ -0,0 +1,240 @@
+
+        // fetch secrets
+        secretClient = SecretManagerResourceManager.builder(PROJECT, credentialsProvider).build();
+        String privateHost = "10.33.0.4";
+        // secretClient.accessSecret(
+        //     "projects/269744978479/secrets/end-to-end-private-ip/versions/1");
+        String publicHost = "34.170.126.170";
+        // secretClient.accessSecret(
+        //     "projects/269744978479/secrets/end-to-end-public-ip/versions/1");
+        String username = "user2";
+        // secretClient.accessSecret("projects/269744978479/secrets/end-to-end-user/versions/1");
+        String password = "password";
+        // secretClient.accessSecret(
+        //     "projects/269744978479/secrets/end-to-end-password/versions/1");
+        String network = "end-to-end";
+        // secretClient.accessSecret("projects/269744978479/secrets/network-mysql/versions/1");
+        String subnetwork = "regions/us-central1/subnetworks/e2e";
+        // secretClient.accessSecret("projects/269744978479/secrets/subnetwork-mysql/versions/1");
+        // create MySql Resources
+        cloudSqlResourceManager = CloudMySQLResourceManager.builder(testName).build();
+        jdbcSource =
+            createMySqlDatabase(
+                cloudSqlResourceManager,
+                new HashMap<>() {
+                  {
+                    put(TABLE, AUTHOR_TABLE_COLUMNS);
+                  }
+                });
+
+        // create and upload GCS Resources
+        gcsResourceManager =
+            GcsResourceManager.builder(artifactBucketName, getClass().getSimpleName(), credentials)
+                .build();
+        createAndUploadShardConfigToGcs(
+            gcsResourceManager, cloudSqlResourceManager, cloudSqlResourceManager.getHost());
+        createAndUploadJarToGcs(gcsResourceManager);
+        gcsResourceManager.createArtifact(
+            "input/session.json",
+            generateSessionFile(
+                cloudSqlResourceManager.getDatabaseName(),
+                spannerResourceManager.getDatabaseId(),
+                SESSION_FILE_RESOURCE));
+
+        // create pubsub manager
+        pubsubResourceManager = setUpPubSubResourceManager();
+
+        // launch forward migration template
+        fwdJobInfo =
+            launchFwdDataflowJob(spannerResourceManager, gcsResourceManager, pubsubResourceManager);
+
+        // launch reverse migration template
+        rrJobInfo =
+            launchRRDataflowJob(
+                spannerResourceManager,
+                gcsResourceManager,
+                spannerMetadataResourceManager,
+                pubsubResourceManager,
+                MYSQL_SOURCE_TYPE,
+                network,
+                subnetwork);
+      }
+    }
+  }
+
+  /**
+   * Cleanup dataflow job and all the resources and resource managers.
+   *
+   * @throws IOException
+   */
+  @AfterClass
+  public static void cleanUp() throws IOException {
```

Comment:
Please check and make sure that there are no resource leaks. All the resourceManagers which are created should be cleaned up.